src/rulenode_operators.jl

```diff
@@ -178,6 +178,13 @@ function _rulenode2expr(typ::Symbol, rulenode::RuleNode, grammar::Grammar, j=0)
 end
 
 
+"""
+Calculates the probability associated with a rulenode in a probabilistic grammar.
+"""
+function rulenode_probability(node::RuleNode, grammar::Grammar)
+	probability(grammar, node.ind) * prod((rulenode_probability(c, grammar) for c ∈ node.children), init=1)
```

Comment:
we should turn the product of probabilities into a sum of log probabilities. This is numerically more stable. We are deemed to run into numerical instability with any medium-sized expression